### PR TITLE
Remove code workarounds necessary for Spark 1.2.1/Hadoop 1.0.x support

### LINKED ISSUE #959: Remove code workarounds necessary for Spark 1.2.1/Hadoop 1.0.x support

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #959 

Works for me locally but should be tested on a cluster over HDFS.

### CODE REVIEW

**Comment by fnothaft** [comment]
LGTM!

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
Let me see if the test failures repro locally for me.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala [hunk 1]
```diff
@@ -389,40 +385,13 @@ private[rdd] class AlignmentRecordRDDFunctions(val rdd: RDD[AlignmentRecord])
...
 
-        // invoke the method on the fs instance
-        // if we are on hadoop 2.x, this makes the call:
-        //
-        // fs.concat(headPath, tailFiles)
         try {
-          concatMethod.invoke(fs, newHeadPath, tailFiles)
+          fs.concat(newHeadPath, tailFiles)
         } catch {
           case ite: InvocationTargetException => {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala]: This needs to change to `case t: Throwable => {`

A line further down will then need to change from `throw ite.getTargetException` to `throw t`

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by heuermh** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
Merged as https://github.com/bigdatagenomics/adam/commit/072f97df75f075d4c298624b84842f8abdb0eeed. Thanks @heuermh!

**Pull request closed without merging**